text/0045-add-notion-of-protocol.md

```diff
@@ -0,0 +1,143 @@
+- `const char **(*get_supported_audio_codecs)(void *data)`: audio codecs supported by the service. Optional, fallback to protocol supported codecs if not set.
+
+Adding to the API these functions:
+- `const char *obs_service_get_protocol(const obs_service_t *service)`: return the protocol used by the service.
+- `obs_service_get_supported_video_codecs(const obs_service_t *service)`: return video codecs compatible with the service.
+- `obs_service_get_supported_audio_codecs(const obs_service_t *service)`: return audio codecs compatible with the service.
+
+### Services and information
+
+Depending on the protocol, the service should provide various types of connection details (e.g. server URL, stream key, username, password).
+
+Currently, `get_key` can provide a stream id (SRT) or an encryption passphrase (RIST) rather than a stream key.
+
+Rather than adding a getter for each possible type of information, a getter where we can choose which type of information we want should be implemented.
+
+Each types of connection details will be define by a macro with an even number, odd number will be reserved for potential third-party protocol (RFC 39).
+
+List of types:
+```c
+#define OBS_SERVICE_INFO_SERVER_URL 0
+#define OBS_SERVICE_INFO_STREAM_KEY 2
+#define OBS_SERVICE_INFO_USERNAME 4 
+#define OBS_SERVICE_INFO_PASSWORD 6
+#define OBS_SERVICE_INFO_STREAM_ID 8
+#define OBS_SERVICE_INFO_ENCRYPT_PASSPHRASE 10
+```
+
+Adding to `obs_service_info`:
+- `const char *(*get_connect_info)(uint32_t type, void *data)` where `type` is one of the value of the list above indicating which info we want and return `NULL` if it doesn't have it.
+- `bool (*can_try_to_connect)(void *data)`, since protocols and services do not always require the same informations. This function allows the service to return if it can connect. Return true if the the service has all it needs to connect.
+
+Adding to the API these functions:
+- `const char *obs_service_get_connect_info(uint32_t type, const obs_service_t *service)`: return the connection information related to the given type (list above).
+- `bool obs_service_can_try_to_connect(const obs_service_t *service)`: return if the service can try to connect.
+  - return `bool (*can_try_to_connect)(void *data)` result.
+  - return true if `bool (*can_try_to_connect)(void *data)` is not implemented in the service.
+  - return false if the service object is invalid.
+
+`obs_service_get_url()`, `obs_service_get_key()`, `obs_service_get_username()`, and `obs_service_get_password()` will be deprecated in favor of `obs_service_get_connect_info()`.
+
+In the future, if we want to support a protocol that doesn't use a server URL, this scenario is covered by this design.
+
+### About `rtmp-services`
+
+`rtmp-services` is plugin that provides two services `rtmp_common` and `rtmp_custom`. The use of "rtmp" in the naming no longer means that it only supports RTMP, it was and is kept to avoid breakage.
```

Comment:
"rtmp_common" and "rtmp_custom" mostly, so "rtmp-services" plugin in general.
RFC 39 plan to replace it.

Reply:
It was mostly a question of if:
> The use of "rtmp" in the naming

applies to just "rtmp-services" or also to "rmtp_common" and "rtmp_custom". I just wanted that clarified in the wording.

Reply:
Resolved with updated wording:
> `rtmp-services` is a plugin that provides two services `rtmp_common` and `rtmp_custom`. The use of "rtmp" in these three terms no longer means that it only supports RTMP, it was and is kept to avoid breakage.